lib/engine.py

```diff
@@ -564,12 +566,15 @@ def format_insert_value(self, value, datatype, escape=True):
             else:
                 return "null"
         elif datatype in ("double", "decimal"):
-            if strvalue:
+
+            if strvalue and strvalue.strip() != "":
```

Comment:
1. If we're going to keep it this way, what about: `if strvalue.strip():`. Seems like it should do the same thing.
2. `strvalue` is already stripped on the 2nd line of the function. It seems like the only place for that to have been lost is when removing extra quotes using `strvalue = strvalue[1:-1]`. So it might be clearner to replace `strvalue = strvalue[1:-1]` with `strvalue = strvalue[1:-1].strip()` and leave this line as it is.